src/coreclr/jit/targetwasm.h

```diff
@@ -0,0 +1,284 @@
+#define FEATURE_TAILCALL_OPT     0       // opportunistic Tail calls (i.e. without ".tail" prefix) made as fast tail calls.
+#define FEATURE_IMPLICIT_BYREFS       1  // Support for struct parameters passed via pointers to shadow copies
+#define FEATURE_MULTIREG_ARGS_OR_RET  0  // Support for passing and/or returning single values in more than one register
+#define FEATURE_MULTIREG_ARGS         0  // Support for passing a single argument in more than one register
+#define FEATURE_MULTIREG_RET          0  // Support for returning a single value in more than one register
+#define MAX_PASS_SINGLEREG_BYTES      8  // Maximum size of a struct passed in a single register (long/double).
+#define MAX_PASS_MULTIREG_BYTES       0  // Maximum size of a struct that could be passed in more than one register
+#define MAX_RET_MULTIREG_BYTES        0  // Maximum size of a struct that could be returned in more than one register (Max is an HFA or 2 doubles)
+#define MAX_ARG_REG_COUNT             1  // Maximum registers used to pass a single argument in multiple registers.
+#define MAX_RET_REG_COUNT             1  // Maximum registers used to return a value.
+#define MAX_MULTIREG_COUNT            2  // Maximum number of registers defined by a single instruction (including calls).
+                                         // This is also the maximum number of registers for a MultiReg node.
+
+#define NOGC_WRITE_BARRIERS      0       // No specialized WriteBarrier JIT Helpers
+#define USER_ARGS_COME_LAST      1
+#ifdef TARGET_WASM32
+#define TARGET_POINTER_SIZE      4       // equal to sizeof(void*) and the managed pointer size in bytes for this target
+#else
+#define TARGET_POINTER_SIZE      8       // equal to sizeof(void*) and the managed pointer size in bytes for this target
+#endif
+#define ETW_EBP_FRAMED           0       // No frame pointer chaining on WASM
+
+// TODO-WASM-CQ: measure if "CSE_CONSTS" is beneficial.
+#define CSE_CONSTS               1       // Enable if we want to CSE constants
+#define EMIT_TRACK_STACK_DEPTH   1       // TODO-WASM: set to 0.
+#define EMIT_GENERATE_GCINFO     0       // Codegen and emit not responsible for GC liveness tracking and GCInfo generation
+
+// Since we don't have a fixed register set on WASM, we set most of the following register defines to 'none'-like values.
+#define REG_FP_FIRST             REG_NA
+#define REG_FP_LAST              REG_NA
+#define FIRST_FP_ARGREG          REG_NA
+#define LAST_FP_ARGREG           REG_NA
+
+#define HAS_FIXED_REGISTER_SET   0       // WASM has an unlimited number of locals/registers.
+#define REGNUM_BITS              1       // number of bits in a REG_*
+#define REGSIZE_BYTES            TARGET_POINTER_SIZE // number of bytes in one general purpose register
+#define FP_REGSIZE_BYTES         8      // number of bytes in one FP/SIMD register
+#define FPSAVE_REGSIZE_BYTES     8       // number of bytes in one FP/SIMD register that are saved/restored, for callee-saved registers
```

Comment:
I'm guessing this isn't 16 because "FP/SIMD register" is referring to architectures like arm64 where there are wide dual-purpose fp/simd registers and narrower integer registers. And for us, we will treat 16-byte SIMD values/locals separately from floats? On the other hand, if our only register (see earlier) is SP, the size of that is 4 in wasm32 and 8 in wasm64, not always 8. And SP isn't a FP/SIMD register, which would mean that we then have no FP/SIMD registers at all...

Reply:
This value is basically only relevant to ARM64 right now, where the calling convention specifies that for a 128 bit vector register that is callee-save, the callee only needs to preserve the lower 8 bytes. So it doesn't matter what we put here.

Since our "registers" are WASM locals, and since all WASM locals are callee-save, `8` absent SIMD128 is as sensible a number as any other.